src/Rules/Classes/EnumSanityRule.php

```diff
@@ -187,6 +176,16 @@ public function processNode(Node $node, Scope $scope): array
 			}
 
 			$exprType = $scope->getType($stmt->expr);
+			$constantValues = $exprType->getConstantScalarValues();
+			if (count($constantValues) === 1) {
```

Comment:
By setting enum cases value here, instead of in the condition
```
if ($stmt->expr instanceof Node\Scalar\Int_ || $stmt->expr instanceof Node\Scalar\String_)
```
I also catch other constant values like the one set with a class Constant.

Cf test
```
enum Backed: int {
	case One = Foo::A;
	case Two = Foo::A;
}
```